zio-http/src/main/scala/zio/http/codec/internal/EncoderDecoder.scala

```diff
@@ -324,34 +448,62 @@ private[codec] object EncoderDecoder                   {
     private def encodeBody(inputs: Array[Any]): Body = {
       if (isByteStream(flattened.content)) {
         Body.fromStream(inputs(0).asInstanceOf[ZStream[Any, Nothing, Byte]])
-      } else if (jsonEncoders.length == 0) Body.empty
-      else if (jsonEncoders.length == 1) {
-        val encoder = jsonEncoders(0)
+      } else {
+        if (inputs.length > 1) {
+          Body.fromMultipartForm(encodeMultipartFormData(inputs), formBoundary)
+        } else {
+          if (jsonEncoders.length == 0) Body.empty
+          else if (jsonEncoders.length == 1) {
+            val encoder = jsonEncoders(0)
+
+            encoder(inputs(0))
+          } else throw new IllegalStateException("A request on a REST endpoint should have at most one body")
+        }
+      }
+    }
 
-        encoder(inputs(0))
-      } else throw new IllegalStateException("A request on a REST endpoint should have at most one body")
+    private def encodeMultipartFormData(inputs: Array[Any]): Form = {
+      Form(
+        flattened.content.zipWithIndex.map { case (bodyCodec, idx) =>
+          val input = inputs(idx)
+          val name  = bodyCodec.name.getOrElse("field" + idx.toString)
+          bodyCodec match {
+            case BodyCodec.Multiple(schema, mediaType, _) if schema == Schema[Byte] =>
+              FormField.streamingBinaryField(
+                name,
+                input.asInstanceOf[ZStream[Any, Nothing, Byte]],
+                mediaType.getOrElse(MediaType.application.`octet-stream`),
+              )
+            case _                                                                  =>
+              formFieldEncoders(idx)(name, input)
+          }
+        }: _*,
+      )
     }
 
     private def encodeContentType(inputs: Array[Any]): Headers = {
       if (isByteStream(flattened.content)) {
         val mediaType = flattened.content(0).mediaType.getOrElse(MediaType.application.`octet-stream`)
         Headers(Header.ContentType(mediaType))
       } else {
-        val _ = inputs // TODO: Support multiple content types
-        if (jsonEncoders.length == 0) Headers.empty
-        else if (jsonEncoders.length == 1) {
-          val mediaType = flattened.content(0).mediaType.getOrElse(MediaType.application.json)
-          Headers(Header.ContentType(mediaType))
-        } else throw new IllegalStateException("A request on a REST endpoint should have at most one body")
+        if (inputs.length > 1) {
+          Headers(Header.ContentType(MediaType.multipart.`form-data`))
+        } else {
+          if (jsonEncoders.length == 0) Headers.empty
+          else if (jsonEncoders.length == 1) {
+            val mediaType = flattened.content(0).mediaType.getOrElse(MediaType.application.json)
+            Headers(Header.ContentType(mediaType))
+          } else throw new IllegalStateException("A request on a REST endpoint should have at most one body")
```

Comment:
This branch should be unreachable as far as I can see, so maybe the tests should be against inputs.length and not jsonEncoders.length, so we don't have this unreachable else. Or am I missing something?